src/library_wasmfs.js

```diff
@@ -107,7 +107,11 @@ mergeInto(LibraryManager.library, {
       // TODO: Remove dependency on FS.cwd().
       // User code should not be using FS.cwd().
       // For file preloading, cwd should be '/' to begin with.
-      return '/';
+      return withStackSave(() => {
+        var buffer = allocateUTF8OnStack(path);
```

Comment:
Where is `path` being defined here?  

The argument to `allocateUTF8OnStack` is a number of bytes so I would have expected something liuke `{{{ PATH_MAX }}}` here.

Reply:
@sbc100 Thank you for your feeback. I had some counter queries 
 
Is it possible to pass path argument to cwd() like the below code snippet
cwd: (path) => {
      return withStackSave(() => {
        var buffer = allocateUTF8OnStack(path);
        var result = _wasmfs_getcwd(buffer,buffer.length);
        return UTF8ToString(buffer);
      });
    },
In case it is not allowed how are we expected to add the path to allocateUTF function.
Also wanted to clarify if the PATH_MAX you are refering to, is same as the path I would pass in the FS.cwd() argument.

Reply:
IIUC FS.cwd() doesn't take an argument ever, it just returns the current working directory as a JS string.

Reply:
@tlively @sbc100 Thanks for the clarification and being patient enough to answer my queries. I have made the changes in the latest commit. Kindly review the changes.

Meanwhile I had some few queries 
1. I was trying to understand different implementation of cwd() which made me come to an alternative to stackAlloc which is adding char buffer[PATH_MAX] in _wasmfs_getcwd() function and then passing it to the __syscall_getcwd call. Should I implement with that approach.
2. To compile library_wasmfs.js I had to comment out FS__deps and set {{{ cDefine('S_IRUGO') }}} {{{ cDefine('S_IXUGO') }}} and {{{ cDefine('S_IWUGO') }}} to some dummy integer. Not sure if it the right approach but the build and testing seems to work after these changes. Can you some alternative to this error resolution.

Reply:
1. Using stack space more sense to me in this case since it avoids permanntly allocating the space (you would need to write (since you would need to add *static* to `char buffer[PATH_MAX]` I believe).
2. ` {{{ cDefine('S_IRUGO') }}}` should work.  If they don't then something odd is going on with your checkout.  What errors do you see?

Reply:
@sbc100 Understood and it seems like a better approach to allocate the space. 
I was getting an 'S_IRUGO' not defined error earlier but now suddenly its not occuring. Trying to figure out the root cause for it for future development and testing purposes. I reckon I missed ./emcc --clear-cache command. Will try to raise an issue for the same.
@kripken @kripken @sbc100 Since I have made all the required changes and the build and testing is successful. Kindly suggest me if I need to make any other changes in the code or the testing.
Also I had a curious question on how to make this code merge ready as I can see some of the checks blocked. Is there any documentation to pass all the checks with proper coding standards.

Reply:
@impact-maker 
I had a similar error with S_IRUGO and for me it was caused by settings `WASMFS=TRUE` in settings.js. By leaving `WASMFS=FALSE` and manually setting the flags in the `test_core.py` as shown below fixed the problem for me. 
```
self.emcc_args += ['-sWASMFS']
self.emcc_args += ['-sFORCE_FILESYSTEM']
```